pkg/apis/core/validation/validation.go

```diff
@@ -4841,6 +4937,7 @@ func ValidatePodCreate(pod *core.Pod, opts PodValidationOptions) field.ErrorList
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("nodeName"), "cannot be set until all schedulingGates have been cleared"))
 	}
 	allErrs = append(allErrs, validateSeccompAnnotationsAndFields(pod.ObjectMeta, &pod.Spec, fldPath)...)
+	allErrs = append(allErrs, validateAppArmorAnnotationsAndFieldsMatchOnCreate(pod.ObjectMeta, &pod.Spec, fldPath)...)
```

Comment:
do we also need to call this from ValidatePodTemplateSpec so pod-spec-containing things don't get a spec that won't be able to create pods?

Reply:
Hmm, good idea. I shouldn't require both to be set on pod templates, but if both are set, they better be in agreement.